bin/methods/setup.js

```diff
@@ -3,22 +3,44 @@ const path = require('path')
 const fs = require('fs')
 const prompt = require('prompt')
 const log = require('log-cb')
-const pixie = require('pixie')
+const configDefaults = require('../files/configDefaults.json')
 
 const CWD = process.cwd()
-const TEMPLATE_PATH = path.join(__dirname, '../files/cordlrrc.tmpl')
-const TEMPLATE = JSON.parse(fs.readFileSync(TEMPLATE_PATH))
+const packagePath = path.join(CWD, 'package.json')
 
 function setup () {
   prompt.get(['token', 'prefix'], function (err, result) {
     if (err) return log()(err)
 
     // Create and write config
-    const config = pixie.compile(TEMPLATE, result)
+    const config = Object.assign({}, configDefaults, { token: result.token, prefix: result.prefix })
     writeConfig(config)
   }).start()
 }
 
-function writeConfig (content) {
-  fs.writeFile(path.join(CWD, '.cordlrrc'), content, log('Configuration created'))
+function writeConfig (config) {
+  return new Promise((resolve, reject) => {
```

Comment:
I wouldn't even create a promise here, or handle the ending in any way:

 1. `writeConfig` is not used by anything else than this file. (just puts some functionality under a more readable name)
 1. It is the last function called in `setup` before program exits, so we wouldn't need to handle anything after it.
 1. Each method inside `bin/methods/` is ran once before the program exits, so we wouldn't need to handle anything aftwards.

If we did in the future, adding a callback wouldn't be very hard at all.  Which leads me onto a side note:

Right now, we don't have any details about use of promises, but it's worth noting that we don't use them anywhere in the code, so this would end up being one of the only times it appears (excluding dependency APIs where we are forced to use them).  

To be honest: I prefer a factory function + callback style.  I find it turns out just as clean!  Promises are stateful, honestly hard to maintain (I've used them for a long time), have weird behavior in Node, and aren't optimized in V8 yet (last time I checked).  Not to be a stickler, because I used promises for a while, I just don't like maintain them!